business-networks-workflows/src/main/kotlin/net/corda/bn/flows/UpdateCordaIdentityFlow.kt

```diff
@@ -0,0 +1,274 @@
+            val signers = authorisedMembers.filter { it.state.data.isActive() }.map { it.state.data.identity.cordaIdentity }
+            val requiredSigners = signers.map { it.owningKey }
+            val updatedGroup = group.state.data.copy(participants = newParticipants)
+            val builder = TransactionBuilder(notary ?: serviceHub.networkMapCache.notaryIdentities.first())
+                    .addInputState(group)
+                    .addOutputState(updatedGroup)
+                    .addCommand(GroupContract.Commands.Modify(requiredSigners), requiredSigners)
+            builder.verify(serviceHub)
+
+            // collect signatures and finalise transaction
+            val observers = (updatedGroup.participants - ourIdentity).map {
+                serviceHub.identityService.wellKnownPartyFromX500Name(it.nameOrNull())!!
+            }
+            val flowSessions = observers.toSet().map { initiateFlow(it) }
+            val finalisedTransaction = collectSignaturesAndFinaliseTransaction(builder, flowSessions, observers)
+            finalisedTransaction
+        }
+
+        return finalisedTransactions.firstOrNull()
+    }
+
+    private fun getAllBusinessNetworkGroups(networkId: String): List<StateAndRef<GroupState>> {
+        val criteria = QueryCriteria.VaultQueryCriteria(Vault.StateStatus.UNCONSUMED)
+                .and(QueryCriteria.VaultCustomQueryCriteria(builder { GroupStateSchemaV1.PersistentGroupState::networkId.equal(networkId) }))
+        return serviceHub.vaultService.queryBy<GroupState>(criteria).states
+    }
+
+    private fun getMembersAuthorisedToModifyGroups(networkId: String): List<StateAndRef<MembershipState>> {
+        val criteria = QueryCriteria.VaultQueryCriteria(Vault.StateStatus.UNCONSUMED)
+                .and(QueryCriteria.VaultCustomQueryCriteria(builder { MembershipStateSchemaV1.PersistentMembershipState::networkId.equal(networkId) }))
+        return serviceHub.vaultService.queryBy<MembershipState>(criteria).states.filter { it.state.data.canModifyGroups() }
+    }
+}
+
+@InitiatedBy(UpdateGroupMembersIdentityFlow::class)
+private class UpdateGroupMembersIdentityResponderFlow(private val session: FlowSession) : MembershipManagementFlow<Unit>() {
+
+    @Suspendable
+    override fun call() {
+        signAndReceiveFinalisedTransaction(session) {
+            if (it.value !is GroupContract.Commands.Modify) {
+                throw FlowException("Only ModifyCordaIdentity command is allowed")
+            }
+        }
+    }
+}
+
+@InitiatingFlow
+private class UpdateImpactedMembershipsFlow(
+        private val changedMembership: StateAndRef<MembershipState>,
+        private val notary: Party?
+) : MembershipManagementFlow<SignedTransaction?>() {
+
+    @Suppress("UNCHECKED_CAST")
+    @Suspendable
+    override fun call(): SignedTransaction? {
+        val networkId = changedMembership.state.data.networkId
+        val name = changedMembership.state.data.identity.cordaIdentity.name
```

Comment:
nit: Maybe fancier in the spirit of idiomatic Kotlin ;)
```suggestion
        val (networkId, name) = changedMembership.state.data.run { networkId to cordaIdentity.name }
        val networkId = changedMembership.state.data.networkId
        val name = changedMembership.state.data.identity.cordaIdentity.name
```